src/Raven.Server/ServerWide/Commands/ClusterTransactionCommand.cs

```diff
@@ -891,20 +922,76 @@ private static unsafe SingleClusterDatabaseCommand ReadCommand<TTransaction>(Tra
 
         public override object FromRemote(object remoteResult)
         {
-            var errors = new List<ClusterTransactionErrorInfo>();
-            if (remoteResult is BlittableJsonReaderArray array)
+            if (remoteResult is BlittableJsonReaderObject bjro)
+            {
+                return GetResults(bjro);
+            }
+
+            if (remoteResult is BlittableJsonReaderArray bjra)
             {
-                foreach (var o in array)
+                return GetErrors(bjra);
+            }
+
+            return base.FromRemote(remoteResult);
+        }
+
+        internal static ClusterTransactionResult GetResults(BlittableJsonReaderObject bjro)
+        {
+            var results = new ClusterTransactionResult();
+            if (bjro.TryGet(nameof(ClusterTransactionResult.GeneratedResult), out BlittableJsonReaderArray bjra))
+                results.GeneratedResult = GetGeneratedResults(bjra);
+
+            return results;
+        }
+
+        private static DynamicJsonArray GetGeneratedResults(BlittableJsonReaderArray bjra)
+        {
+            var array = new DynamicJsonArray();
+
+            if (bjra != null)
+            {
+                foreach (BlittableJsonReaderObject item in bjra)
                 {
-                    if (o is not BlittableJsonReaderObject blittable)
-                        continue;
+                    array.Add(ToDynamicJsonValue(item));
+                }
+            }
+
+            return array;
+        }
 
-                    errors.Add(ToClusterTransactionErrorInfo(blittable));
+        private static DynamicJsonValue ToDynamicJsonValue(BlittableJsonReaderObject bjro)
+        {
+            var djv = new DynamicJsonValue();
+            var prop = new BlittableJsonReaderObject.PropertyDetails();
+
+            for (int i = 0; i < bjro.Count; i++)
+            {
+                bjro.GetPropertyByIndex(i, ref prop);
+                if (prop.Value is LazyStringValue lsv)
+                {
+                    djv[prop.Name] = lsv.ToString();
+                }
+                else
+                {
+                    var json = prop.Value as IDynamicJson;
+                    djv[prop.Name] = json == null ? (object)prop.Value : json.ToJson();
                 }
+            }
```

Comment:
Would prefer to avoid this and work with `TryGetMember(...)`

Reply:
Done